retrofit/src/main/java/retrofit2/OkHttpCall.java

```diff
@@ -140,10 +138,10 @@ private void callSuccess(Response<T> response) {
     if (code < 200 || code >= 300) {
       try {
         // Buffer the entire body to avoid future I/O.
-        ResponseBody bufferedBody = Utils.readBodyToBytesIfNecessary(rawBody);
+        ResponseBody bufferedBody = Utils.buffer(rawBody);
```

Comment:
Can't use `rawResponse.peekBody` here because we replace its body with a no-content one.